src/extra.ml

```diff
@@ -128,11 +193,47 @@ module Array =
         elements with [eq]. *)
     let equal : 'a eq -> 'a array eq = fun eq a1 a2 ->
       Array.length a1 = Array.length a2 && for_all2 eq a1 a2
+
+    (** [argmax ?cmp a] returns the index of the first maximum of array [a]
+        according to comparison [?cmp].  If [cmp] is not given, defaults to
+        [Pervasives.compare]. *)
+    let argmax : ?cmp:('a -> 'a -> int) -> 'a array -> int =
+      fun ?(cmp=Pervasives.compare) arr ->
+      if arr = [||] then invalid_arg "Extra.Array.argmax" else
+      let r, _, _ = Array.fold_left (fun (mi, m, i) elt ->
+        if cmp elt m > 0 then (i, elt, succ i) else (mi, m, succ i))
+        (0, arr.(0), 0) arr in
+      r
+
+    (** [max ?cmp a] returns the higher element according to comparison
+        function [?cmp], using [Pervasives.compare] if not given, in array
+        [a]. *)
+    let max : ?cmp:('a -> 'a -> int)-> 'a array -> 'a =
+      fun ?(cmp=Pervasives.compare) arr ->
+      if arr = [||] then invalid_arg "Extra.Array.max" else
+      Array.fold_left (fun acc elt -> if cmp elt acc >= 0 then elt else acc)
+        arr.(0) arr
+
+    (** [split a] is {!val:List.split}[Array.to_list a]. *)
```

Comment:
Rather:
```ocaml
(** [split ar] is a more efficient version of [List.split (Array.to_list ar)]. *)
```
Also the parentheses are useless in the return type.
And please, use a local `let` to define the folding function.